packages/foam-vscode/src/core/markdown-provider.ts

```diff
@@ -201,6 +222,53 @@ const tagsPlugin: ParserPlugin = {
   },
 };
 
+let blockStack: Array<{ label: string; level: number; start: Position }> = [];
+const blocksPlugin: ParserPlugin = {
+  name: 'block',
+  onWillVisitTree: () => {
+    blockStack = [];
+  },
+  visit: (node, note) => {
+    if (node.type === 'heading') {
+      const level = (node as any).depth;
+      const label = ((node as Parent)!.children?.[0] as any)?.value;
```

Comment:
Not sure I follow, how would you change the line? 
I admit sometimes I get lost in TS's types so I have overdone it haha

Reply:
(anyway it's done this way in other places so it's probably better to leave it alone just for the sake of consistency, but would you mind explaining it?)

Reply:
I'm just learning about the type system as we speak but I think that once you say `x is T` you don't need to also signal the compiler it can't be null by adding the `!`. Consider this example:

```ts
interface InnerThing {
    realThing: number;
}

interface Thing {
    innerThing?: InnerThing;
}

function doStuffWith(aThing: Thing, anotherThing: Thing) : number {
    return (aThing.innerThing as InnerThing).realThing + anotherThing.innerThing!.realThing;
}
```

When  compiled with strict null checking the last line can't just be  `aThing.innerThing.realThing + anotherThing.innerThing.realThing`, but  both  doing the cast  or adding the `!` would be enough to silence the compiler, you don't need to do both of them.

That said, it doesn't do any harm and since it's consistently used in a similar way at other locations maybe it's better not to touch it.